backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/storage/backup/HybridBackupCommand.java

```diff
@@ -175,6 +176,7 @@ public boolean performNextOperation(int completedChildCount) {
 
     @Override
     protected void endSuccessfully() {
+        removeAutoGeneratedSnapshot(cloneContextAndDetachFromParent(), null, null);
```

Comment:
I think we need to wrap this with a check if the snapshot exists, maybe worth checking if it's initiated because the backup was stopped early for logging purposes

Reply:
The thing is, the snapshot creation is done in executeCommand, and by the time the backup id is returned (and backup can be stopped) - the snapshot already exists.

Reply:
yes, but the snapshot can be removed inside performNextOperation (normal flow), so it will try to remove it again no?

Reply:
I have removed snapshot removal from performNextOperation as you suggested, now we never wait for it to finish. But added a check that snapshot exists